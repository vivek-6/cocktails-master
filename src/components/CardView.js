import React from "react";
import CocktailCard from "./CocktailCard";
import { GridList } from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";

const styles = () => {
  return {
    gridList: {
      justifyContent: "center"
    }
  };
};

const CardView = ({ classes, displayedCocktails }) => {
  return (
    <GridList className={classes.gridList}>
      {displayedCocktails.map(cocktail => (
        <CocktailCard key={cocktail.name} cocktail={cocktail} />
      ))}
    </GridList>
  );
};

export default withStyles(styles)(CardView);
