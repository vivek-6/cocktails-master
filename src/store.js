import appReducer from "./reducers";
import { createStore, applyMiddleware, compose } from "redux";

import {
  persistCurrentState,
  supportsPersistence
} from "./utilities/persistence";
import throttle from "lodash/throttle";

import thunk from "redux-thunk";

const devtools =
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();

const middlewares = compose(
  applyMiddleware(thunk),
  devtools || (a => a)
);

const store = createStore(appReducer, middlewares);

// persist parts of the store whenever it changes.
if (supportsPersistence()) {
  store.subscribe(
    throttle(() => {
      persistCurrentState(store.getState(), ["bar", "settings", "favourites"]);
    })
  );
}

export default store;
