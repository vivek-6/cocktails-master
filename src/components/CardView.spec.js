import React from "react";
import renderer from "react-test-renderer";
import CardView from "./CardView";
import store from "../store";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import { loadDatabase } from "../utilities/db.utils";

beforeAll(async () => {
  await loadDatabase(store);
});

it("does not explode when rendered", async () => {
  const displayedCocktails = [store.getState().db.cocktails[0]];

  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <CardView displayedCocktails={displayedCocktails} />
      </MemoryRouter>
    </Provider>
  );

  expect(tree).toMatchSnapshot();
});
