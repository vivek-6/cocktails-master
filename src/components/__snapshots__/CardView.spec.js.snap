// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not explode when rendered 1`] = `
<ul
  className="MuiGridList-root CardView-gridList-1"
  style={
    Object {
      "margin": -2,
    }
  }
>
  <div
    className="MuiPaper-root MuiCard-root CocktailCard-card-3 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiCardActionArea-root CocktailCard-cardMain-4"
      href="/cocktails/vesper"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiCardHeader-root"
      >
        <div
          className="MuiCardHeader-avatar"
        >
          <div
            aria-label="Recipe"
            className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            style={
              Object {
                "background": "linear-gradient(#D88317)",
                "backgroundColor": "#D88317",
              }
            }
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <svg>
                Martini.svg
              </svg>
            </svg>
          </div>
        </div>
        <div
          className="MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
          >
            <h1
              className="CocktailCard-title-6"
            >
              Vesper
            </h1>
          </span>
          <span
            className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            <span
              className="CocktailCard-subHeader-7"
            >
              Before Dinner Cocktail
            </span>
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root"
      >
        <ul
          className="CocktailCard-ingredientList-2"
        >
          <li>
            <p
              className="MuiTypography-root MuiTypography-body1"
            >
              <span>
                6 cl
                 
                <strong
                  aria-describedby={null}
                  className="Connect(IngredientDetail)-tooltip-10"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  title="40% abv"
                >
                  Gin
                </strong>
              </span>
            </p>
          </li>
          <li>
            <p
              className="MuiTypography-root MuiTypography-body1"
            >
              <span>
                1.5 cl
                 
                <strong
                  aria-describedby={null}
                  className="Connect(IngredientDetail)-tooltip-10"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  title="40% abv"
                >
                  Vodka
                </strong>
              </span>
            </p>
          </li>
          <li>
            <p
              className="MuiTypography-root MuiTypography-body1"
            >
              <span>
                0.75 cl
                 
                <strong
                  aria-describedby={null}
                  className="Connect(IngredientDetail)-tooltip-10"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  title="15% abv"
                >
                  Lillet Blonde
                </strong>
              </span>
            </p>
          </li>
        </ul>
        <br />
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <svg>
              Martini.svg
            </svg>
          </svg>
           
          Martini
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18.4 10.6C16.55 8.99 14.15 8 11.5 8c-4.65 0-8.58 3.03-9.96 7.22L3.9 16c1.05-3.19 4.05-5.5 7.6-5.5 1.95 0 3.73.72 5.12 1.88L13 16h9V7l-3.6 3.6z"
            />
          </svg>
           
          Lemon twist
        </p>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
      <span
        className="MuiTouchRipple-root"
      />
    </a>
    <div
      className="MuiCardActions-root CocktailCard-actions-5 MuiCardActions-spacing"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text CocktailActions-button-11 MuiButton-textSecondary MuiButton-textSizeLarge MuiButton-sizeLarge"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
            />
          </svg>
          Favourite
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-text CocktailActions-button-11 MuiButton-textSecondary MuiButton-textSizeLarge MuiButton-sizeLarge"
        href="/cocktails/vesper"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 5c-1.11-.35-2.33-.5-3.5-.5-1.95 0-4.05.4-5.5 1.5-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6v14.65c0 .25.25.5.5.5.1 0 .15-.05.25-.05C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5 1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.1.05.15.05.25.05.25 0 .5-.25.5-.5V6c-.6-.45-1.25-.75-2-1zm0 13.5c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5V8c1.35-.85 3.8-1.5 5.5-1.5 1.2 0 2.4.15 3.5.5v11.5z"
            />
            <path
              d="M17.5 10.5c.88 0 1.73.09 2.5.26V9.24c-.79-.15-1.64-.24-2.5-.24-1.7 0-3.24.29-4.5.83v1.66c1.13-.64 2.7-.99 4.5-.99zM13 12.49v1.66c1.13-.64 2.7-.99 4.5-.99.88 0 1.73.09 2.5.26V11.9c-.79-.15-1.64-.24-2.5-.24-1.7 0-3.24.3-4.5.83zM17.5 14.33c-1.7 0-3.24.29-4.5.83v1.66c1.13-.64 2.7-.99 4.5-.99.88 0 1.73.09 2.5.26v-1.52c-.79-.16-1.64-.24-2.5-.24z"
            />
          </svg>
          Learn More
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</ul>
`;
